tests/ballerina-test/src/test/java/org/ballerinalang/test/services/configuration/OverflowConfigurationTest.java

```diff
@@ -0,0 +1,133 @@
+package org.ballerinalang.test.services.configuration;
+
+import org.ballerinalang.launcher.util.BServiceUtil;
+import org.ballerinalang.launcher.util.CompileResult;
+import org.ballerinalang.mime.util.MimeUtil;
+import org.ballerinalang.nativeimpl.io.channels.base.Channel;
+import org.ballerinalang.nativeimpl.io.channels.base.CharacterChannel;
+import org.ballerinalang.test.nativeimpl.functions.io.MockByteChannel;
+import org.ballerinalang.test.nativeimpl.functions.io.util.TestUtil;
+import org.ballerinalang.test.services.testutils.HTTPTestRequest;
+import org.ballerinalang.test.services.testutils.MessageUtils;
+import org.ballerinalang.test.services.testutils.Services;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.Assert;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+import org.wso2.transport.http.netty.message.HTTPCarbonMessage;
+import org.wso2.transport.http.netty.message.HttpMessageDataStreamer;
+
+import java.io.ByteArrayOutputStream;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URISyntaxException;
+import java.nio.channels.ByteChannel;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+public class OverflowConfigurationTest {
+    private static final Logger log = LoggerFactory.getLogger(OverflowConfigurationTest.class);
+
+    private static final String MOCK_ENDPOINT_NAME = "mockEP";
+    private CompileResult serviceResult;
+
+    @BeforeClass
+    public void setup() {
+        String sourceFilePath = "test-src/services/configuration/overflow_annotation_test.bal";
+        serviceResult = BServiceUtil.setupProgramFile(this, sourceFilePath);
+    }
+
+    @Test(description = "Test default overflow settings (memory threshold 2MB) without actual overflow")
+    public void testDefaultWithoutOverflow() {
+        String path = "/defaultOverflow";
+        try {
+            String content = "ballerina";
+            HTTPTestRequest cMsg = MessageUtils.generateHTTPMessage(path, "POST", content);
+            HTTPCarbonMessage response = Services.invokeNew(serviceResult, MOCK_ENDPOINT_NAME, cMsg);
+            Assert.assertNotNull(response, "Response message not found");
+            InputStream inputStream = new HttpMessageDataStreamer(response).getInputStream();
+            Assert.assertNotNull(inputStream, "Inputstream is null");
+            ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+            MimeUtil.writeInputToOutputStream(inputStream, outputStream);
+            Assert.assertEquals(outputStream.size(), content.getBytes().length);
```

Comment:
Above code block is duplicated in multiple places in this file. Can we add a new method for that block?